gcloud/storage/blob.py

```diff
@@ -266,7 +286,8 @@ def delete(self, client=None):
                  (propagated from
                  :meth:`gcloud.storage.bucket.Bucket.delete_blob`).
         """
-        return self.bucket.delete_blob(self.name, client=client)
+        return self.bucket.delete_blob(self.name, client=client,
+                                       generation=self.generation)
```

Comment:
I had forgotten we give such little info to` Bucket.delete_blob` (i.e. it only has the blob's name, not the object itself).